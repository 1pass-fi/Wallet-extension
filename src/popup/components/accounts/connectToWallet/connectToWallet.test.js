import '@babel/polyfill'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { BrowserRouter as Router } from 'react-router-dom'

jest.mock('utils')

import ConnectToWallet from '.'

const mockStore = configureStore([])

describe('Test for ConnectToWallet component', () => {
  let setErrorMock, storeMock, container

  beforeEach(() => {
    setErrorMock = jest.fn()

    const initState = {
      setError: setErrorMock,
    }

    storeMock = mockStore(initState)

    container = render(
      <Router>
        <Provider store={storeMock}>
          <ConnectToWallet />
        </Provider>
      </Router>
    ).container
  })

  describe('renders without crashing', () => {
    it('renders correctly', () => {
      expect(container).toMatchSnapshot()
    })
  })

  describe('click on check all button and clear checked', () => {
    it('renders correctly', () => {
      const selectAllButton = document.querySelector('.select-all-button')
      const clearAllButton = document.querySelector('.unselect-button')
      const optionButton = document.querySelector('.check-wallet')
      fireEvent.click(selectAllButton)
      expect(container).toMatchSnapshot()
      fireEvent.click(clearAllButton)
      expect(container).toMatchSnapshot()
      fireEvent.click(optionButton)
      expect(container).toMatchSnapshot()
    })
  })

  describe('click on reject button', () => {
    it('renders correctly', () => {
      const rejectButton = document.querySelector('.button.reject')
      fireEvent.click(rejectButton)
      expect(container).toMatchSnapshot()
    })
  })

  describe('click on connect button', () => {
    it('renders correctly', () => {
      const connectButton1 = document.querySelector('.button.connect')
      fireEvent.click(connectButton1)

      const connectButton2 = document.querySelector(
        '.allow-permission .button.connect'
      )
      fireEvent.click(connectButton2)
      expect(container).toMatchSnapshot()
    })
  })
})
