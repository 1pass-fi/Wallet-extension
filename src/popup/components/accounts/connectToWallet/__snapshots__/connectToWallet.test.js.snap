// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ConnectToWallet component click on check all button and clear checked renders correctly 1`] = `
<div>
  <div
    class="select-wallet"
  >
    <header
      class="header"
    >
      <div
        class="connect-with-koi"
      >
        Connect with Koi
      </div>
      <div
        class="title"
      >
        
        <a
          class="company-url"
          href=""
          target="_blank"
        />
      </div>
    </header>
    <div
      class="content"
    >
      <div
        class="card card-content"
      >
        <div
          class="select-account"
        >
          Select accounts
        </div>
        <div
          class="wallet-options"
        >
          <div
            class="wallet-option false"
          >
            <div
              class="checkbox-container check-wallet"
            >
              <div
                class="checkbox-input"
              >
                <input
                  checked=""
                  class="checkbox-hidden"
                  disabled=""
                  type="checkbox"
                />
                <div
                  class="checkbox-styled"
                  style="background: rgb(155, 231, 196);"
                >
                  <div
                    class="check-mark-icon"
                    style="visibility: visible;"
                  >
                    <svg />
                  </div>
                </div>
              </div>
            </div>
            <svg
              class="wallet-icon"
            />
            <div
              class="wallet-info"
            >
              <div
                class="wallet-name"
              />
              <div
                class="wallet-address"
              >
                
                ...
                
              </div>
            </div>
          </div>
        </div>
        <div
          class="trust-sites"
        >
          Only connect with sites you trust.
        </div>
        <div
          class="button-line"
        >
          <button
            class="button-shared button-filled button connect"
          >
            Connect
          </button>
          <button
            class="button-shared button-outline button reject"
          >
            Reject
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test for ConnectToWallet component click on connect button renders correctly 1`] = `
<div>
  <div
    class="select-wallet"
  >
    <header
      class="header"
    >
      <div
        class="connect-with-koi"
      >
        Connect with Koi
      </div>
      <div
        class="title"
      >
        
        <a
          class="company-url"
          href=""
          target="_blank"
        />
      </div>
    </header>
    <div
      class="content"
    >
      <div
        class="card card-content"
      >
        <div
          class="allow-permission"
        >
          <div
            class="label"
          >
            Allow this site to:
          </div>
          <div
            class="permission checked"
          >
            <div
              class="checkbox-container check-allow"
            >
              <div
                class="checkbox-input"
              >
                <input
                  checked=""
                  class="checkbox-hidden"
                  disabled=""
                  type="checkbox"
                />
                <div
                  class="checkbox-styled"
                  style="background: rgb(155, 231, 196);"
                >
                  <div
                    class="check-mark-icon"
                    style="visibility: visible;"
                  >
                    <svg />
                  </div>
                </div>
              </div>
              <span
                class="checkbox-label"
              >
                View the addresses of your permitted accounts (required)
              </span>
            </div>
          </div>
          <div
            class="trust-sites"
          >
            Only connect with sites you trust.
          </div>
          <div
            class="button-line"
          >
            <button
              class="button-shared button-filled button connect"
            >
              Connect
            </button>
            <button
              class="button-shared button-outline button reject"
            >
              Reject
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test for ConnectToWallet component click on reject button renders correctly 1`] = `
<div>
  <div
    class="select-wallet"
  >
    <header
      class="header"
    >
      <div
        class="connect-with-koi"
      >
        Connect with Koi
      </div>
      <div
        class="title"
      >
        
        <a
          class="company-url"
          href=""
          target="_blank"
        />
      </div>
    </header>
    <div
      class="content"
    >
      <div
        class="card card-content"
      >
        <div
          class="select-account"
        >
          Select accounts
        </div>
        <div
          class="wallet-options"
        >
          <div
            class="wallet-option false"
          >
            <div
              class="checkbox-container check-wallet"
            >
              <div
                class="checkbox-input"
              >
                <input
                  checked=""
                  class="checkbox-hidden"
                  disabled=""
                  type="checkbox"
                />
                <div
                  class="checkbox-styled"
                  style="background: rgb(155, 231, 196);"
                >
                  <div
                    class="check-mark-icon"
                    style="visibility: visible;"
                  >
                    <svg />
                  </div>
                </div>
              </div>
            </div>
            <svg
              class="wallet-icon"
            />
            <div
              class="wallet-info"
            >
              <div
                class="wallet-name"
              />
              <div
                class="wallet-address"
              >
                
                ...
                
              </div>
            </div>
          </div>
        </div>
        <div
          class="trust-sites"
        >
          Only connect with sites you trust.
        </div>
        <div
          class="button-line"
        >
          <button
            class="button-shared button-filled button connect"
          >
            Connect
          </button>
          <button
            class="button-shared button-outline button reject"
          >
            Reject
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test for ConnectToWallet component renders without crashing renders correctly 1`] = `
<div>
  <div
    class="select-wallet"
  >
    <header
      class="header"
    >
      <div
        class="connect-with-koi"
      >
        Connect with Koi
      </div>
      <div
        class="title"
      >
        
        <a
          class="company-url"
          href=""
          target="_blank"
        />
      </div>
    </header>
    <div
      class="content"
    >
      <div
        class="card card-content"
      >
        <div
          class="select-account"
        >
          Select accounts
        </div>
        <div
          class="wallet-options"
        >
          <div
            class="wallet-option false"
          >
            <div
              class="checkbox-container check-wallet"
            >
              <div
                class="checkbox-input"
              >
                <input
                  checked=""
                  class="checkbox-hidden"
                  disabled=""
                  type="checkbox"
                />
                <div
                  class="checkbox-styled"
                  style="background: rgb(155, 231, 196);"
                >
                  <div
                    class="check-mark-icon"
                    style="visibility: visible;"
                  >
                    <svg />
                  </div>
                </div>
              </div>
            </div>
            <svg
              class="wallet-icon"
            />
            <div
              class="wallet-info"
            >
              <div
                class="wallet-name"
              />
              <div
                class="wallet-address"
              >
                
                ...
                
              </div>
            </div>
          </div>
        </div>
        <div
          class="trust-sites"
        >
          Only connect with sites you trust.
        </div>
        <div
          class="button-line"
        >
          <button
            class="button-shared button-filled button connect"
          >
            Connect
          </button>
          <button
            class="button-shared button-outline button reject"
          >
            Reject
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
